pkg/workflows/workflow.go

```diff
@@ -566,3 +583,35 @@ func (w *workflowLifecycle) injectActiveDeadlineSeconds(wf *unstructured.Unstruc
 
 	return nil
 }
+
+func IsValidV1WorkFlow(obj interface{}) error {
```

Comment:
Let's not blindly accept `interface{}` object. Use controller-runtime `Object` interface instead:
```suggestion
func IsValidV1WorkFlow(obj runtime.Object) error {
```

Then we don't need to do type assertion at all.

Reply:
a61e54f29fc651d349b0b4f6c853ed6ef6c615cd
done